twilight-gateway/src/config.rs

```diff
@@ -571,15 +592,33 @@ mod tests {
             ConfigBuilder::new(WITHOUT.to_owned(), Intents::empty())
                 .build()
                 .token
+                .inner
                 .as_ref(),
             WITH
         );
         assert_eq!(
             ConfigBuilder::new(WITH.to_owned(), Intents::empty())
                 .build()
                 .token
+                .inner
                 .as_ref(),
             WITH
         );
     }
+
+    #[tokio::test]
+    async fn config_debug() {
+        let config = Config::new("Bot foo".to_owned(), Intents::empty());
+
+        assert_eq!(
+            format!("{config:?}"),
+            format!(
+                "Config {{ event_types: {event_types:?}, gateway_url: None, identify_properties: None, intents: {intents:?}, large_threshold: {large_threshold:?}, presence: None, ratelimit_messages: true, queue: {queue:?}, session: None, tls: TlsContainer, token: <redacted> }}",
+                event_types = config.event_types,
+                intents = config.intents,
+                large_threshold = config.large_threshold,
+                queue = config.queue,
+            ),
+        );
+    }
```

Comment:
Unfortunate to depend upon the whole debug implementation.
```suggestion
        assert!(format!("{config:?}").contains("token: <redacted>"))
```